api/environments/dynamodb/wrappers/environment_wrapper.py

```diff
@@ -116,21 +102,33 @@ def get_identity_overrides_by_environment_id(
         except KeyError as e:
             raise ObjectDoesNotExist() from e
 
-    def get_page_of_feature_identities(
+    def get_identity_overrides_page(
         self, environment_id: int, feature_id: int
     ) -> list[dict[str, Any]]:
         query_response = self.table.query(
-            KeyConditionExpression=Key(ENVIRONMENTS_V2_PARTITION_KEY).eq(
-                str(environment_id),
-            )
-            & Key(ENVIRONMENTS_V2_SORT_KEY).begins_with(
-                get_environments_v2_identity_override_document_key(
-                    feature_id=feature_id,
-                ),
+            KeyConditionExpression=self.get_identity_overrides_key_condition_expression(
+                environment_id=environment_id,
+                feature_id=feature_id,
             )
         )
+        last_evaluated_key = query_response.get("LastEvaluatedKey")
+        for item in query_response["Items"]:
+            item["more_identity_overrides"] = last_evaluated_key is not None
```

Comment:
I understand the desire to preserve the function return type, but this attribute does not belong to individual items.

Let's return a typed dict with `items: list[dict[str, Any]` and a `more_identity_overrides: bool` keys.

I would also suggest to think on the naming some more. I prefer the boolean attribute names to be more apparent, e.g. `is_num_identity_overrides_complete`.

Reply:
As discussed over a call with @khvn26 we're going with a `dataclass` over a `TypedDict` for the return type.